packages/devtools_app/lib/src/debugger/evaluate.dart

```diff
@@ -274,55 +284,232 @@ class _ExpressionEvalFieldState extends State<ExpressionEvalField>
+        if (response.typeClass != null) {
+          // Assume we want static members for a type class not members of the
+          // Type object. This is reasonable as Type objects are rarely useful
+          // in dart.
+          result.addAll(await _autoCompleteMembersFor(
+            response.typeClass,
             controller,
-          ),
-        );
-        // TODO(grouma) - This shouldn't be necessary but package:dwds does
-        // not properly provide superclass information.
-        final clazz = await classFor(instance.classRef, controller);
-        result.addAll(instance.fields
-            .map((field) => field.decl.name)
-            .where((member) => _isAccessible(member, clazz, controller)));
+            staticContext: true,
+          ));
+        } else {
+          await _addAllInstanceMembersToAutocompleteList(
+            result,
+            response,
+            controller,
+          );
+        }
       }
     } catch (_) {}
   }
-  return result.where((name) => name.startsWith(parts.activeWord)).toList()
-    ..sort();
+  return result.where((name) => name.startsWith(parts.activeWord)).toList();
 }
 
-Future<Set<String>> _autoCompleteMembersFor(
-  ClassRef classRef,
+// Due to https://github.com/dart-lang/sdk/issues/46221
+// we cannot tell what the show clause for an export was so it is unsafe to
+// surface exports as if they were library members as there tend to be
+// significant false positives for libraries such as Flutter where all of
+// dart:ui shows up as in scope from flutter:foundation when it should not be.
+bool debugIncludeExports = true;
+
+Future<Set<String>> libraryMemberAndImportsAutocompletes(
+    LibraryRef libraryRef, DebuggerController controller) {
+  return controller.autocompleteInclusiveCache.putIfAbsent(libraryRef,
+      () => _libraryMemberAndImportsAutocompletes(libraryRef, controller));
+}
+
+Future<Set<String>> _libraryMemberAndImportsAutocompletes(
+    LibraryRef libraryRef, DebuggerController controller) async {
+  final result = <String>{};
+  try {
+    final futures = <Future<Set<String>>>[];
+    futures.add(libraryMemberAutocompletes(
+      controller,
+      libraryRef,
+      includePrivates: true,
+    ));
+
+    final Library library = await controller.getObject(libraryRef);
+    for (var dependency in library.dependencies) {
+      if (dependency.prefix?.isNotEmpty ?? false) {
+        // We won't give a list of autocompletes once you enter a prefix
+        // but at least we do include the prefix in the autocompletes list.
+        result.add(dependency.prefix);
+      } else {
+        futures.add(libraryMemberAutocompletes(
+          controller,
+          dependency.target,
+          includePrivates: false,
+        ));
+      }
+    }
+    (await Future.wait(futures)).forEach(result.addAll);
+  } catch (_) {
+    // Silently skip library completions if there is a failure.
+  }
+  return result;
+}
+
+Future<Set<String>> libraryMemberAutocompletes(
```

Comment:
Why does this need to be public?

Reply:
there is a matching helper method with the same name that is private.  Doesn't have to be public but I don't see a lot of harm in making it public.

Reply:
It's just good practice to limit the public interface when possible, even for internal classes, to reduce the difficulty of refactors.